syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.routeguide";
option java_outer_classname = "RouteGuideProto";
option objc_class_prefix = "RTG";

package model;

service Chat {
    rpc client_receive_message(Text) returns (Note) {}
    rpc client_send_message(Note) returns (Text) {}
}

message Text {
    string text = 1;
}
message Note {
    int64 sender = 1;
    int64 recipient = 2;
    int64 logical_clock_time = 3;
    int64 length_queue = 4;
}

